packages/react/src/useFindOne.ts

```diff
@@ -0,0 +1,49 @@
+import { findOneOperation, GadgetRecord, get, hydrateRecord, LimitToKnownKeys, Select } from "@gadgetinc/api-client-core";
+import { useMemo } from "react";
+import { useQuery, UseQueryResponse } from "urql";
+import { FindOneFunction } from "./GadgetFunctions";
+import { OptionsType } from "./OptionsType";
+import { useStructuralMemo } from "./useStructuralMemo";
+
+/**
+ * React hook to fetch a Gadget record using the `findOne` method of a given manager.
+ *
+ * @param manager Gadget model manager to use
+ * @param id id of the record to fetch
+ * @param options options for selecting the fields in the result
+ */
+export const useFindOne = <
+  OptionsT extends OptionsType, // currently necessary for Options to be a narrow type (e.g., `true` instead of `boolean`)
+  F extends FindOneFunction<OptionsT, any, any, any>,
+  Options extends F["optionsType"]
```

Comment:
As you can see, I'm actually not exposing any urql options right now. I think we'd want to do that, eh? Or should we expose our own layer on top of it? I'm inclined to do the former, as I think we're pretty committed to the urql ecosystem.

Reply:
Yeah, I think we should proxy through any urql options the developer wants to set so we don't take away power unnecessarily, so `Options extends F["optionsType"] & Omit<UseQueryArgs<never, never>, 'query' | 'variables'>` or something like that. There's not actually that many other options (see https://github.com/FormidableLabs/urql/blob/main/packages/react-urql/src/hooks/useQuery.ts#L22-L28) so we could even just inline those types here if that omit business seems too complicated.